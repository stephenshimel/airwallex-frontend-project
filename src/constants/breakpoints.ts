const size = {
	mobile: "768px",
	tablet: "992px",
	desktop: "1200px",
};

export const device = {
	mobile: `(max-width: ${size.mobile})`,
	tablet: `(max-width: ${size.tablet})`,
	desktop: `(max-width: ${size.desktop})`,
};
