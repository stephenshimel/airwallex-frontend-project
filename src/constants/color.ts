export const colors = {
	grey: "#CCCBCA",
	lightGrey: "#808080",
	black: "#000000",
	white: "#FFFFFF",
};
