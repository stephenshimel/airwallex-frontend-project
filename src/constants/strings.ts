const companyName = "Broccoli & Co.";

export const strings = {
	// TODO: category Home page
	homePage: {
		header: { title: companyName.toUpperCase() },
		footer: {
			madeFrom: `Made with \u2665 in Melbourne`,
			copyRight: `Â© 2016 Broccoli & Co. All rights reserved`,
		},
		content: {
			homepageAdTitle: "A better way to enjoy every day",
			homepageAdText: "Be the first to know when we launch",
			requestInviteButtonText: "Request an invite",
		},
	},
	requestInvitePage: {
		requestFormTitle: "Request an invite",
		sendButtonLabel: "Send",
		loadingButtonLabel: "Sending, pleasae wait...",
	},
	submitSuccessPage: {
		title: "All done!",
		subTitle: `You will be one of the first to experience ${companyName} when w launch`,
		buttonLabel: "OK",
	},
};
