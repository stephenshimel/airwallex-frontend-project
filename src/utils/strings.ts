export const strings = {
	companyName: "BROCCOLI & CO.",
	madeFrom: `Made with \u2665 in Melbourne`,
	copyRight: `Â© 2016 Broccoli & Co. All rights reserved`,
	homepageAdTitle: "A better way to enjoy every day",
	homepageAdText: "Be the first to know when we launch",
	requestInviteButtonText: "Request an invite",
	requestFormTitle: "Request an invite",
	sendButtonLabel: "Send",
	loadingButtonLabel: "Sending, pleasae wait...",
};
