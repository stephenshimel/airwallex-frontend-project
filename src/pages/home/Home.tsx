import React, { useState } from "react";
import Footer from "../../component/footer/Footer";
import Header from "../../component/header/Header";
import { strings } from "../../utils/strings";
import {
	PageContent,
	PageWrapper,
	AdTitle,
	AdText,
} from "./styles/Home.styles";
import Button from "../../component/button/Button";
import Modal from "../../component/modal/Modal";

import RequestInviteForm from "./request-invite-form/RequestInviteForm";

function Home() {
	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);

	return (
		<PageWrapper isActive={!isModalOpen}>
			<Header title={strings.companyName} />

			<PageContent>
				<AdTitle>{strings.homepageAdTitle}</AdTitle>
				<AdText>{strings.homepageAdText}</AdText>
				<Button
					label={strings.requestInviteButtonText}
					callback={() => {
						setIsModalOpen(true);
					}}
				/>
			</PageContent>

			<Footer texts={[strings.madeFrom, strings.copyRight]} />

			{isModalOpen && (
				<Modal
					content={<RequestInviteForm />}
					closeModal={() => {
						setIsModalOpen(false);
					}}
				/>
			)}
		</PageWrapper>
	);
}

export default Home;
