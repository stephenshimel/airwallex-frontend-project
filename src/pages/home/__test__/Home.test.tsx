import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "../Home";
import { strings } from "../../../constants/strings";
import { QueryClient, QueryClientProvider } from "react-query";

describe("Home", () => {
	const queryClient = new QueryClient();

	it("renders correctly", () => {
		render(<Home />);

		// Check if the static elements are correctly rendered
		expect(screen.getByText(strings.homePage.header.title)).toBeInTheDocument();
		expect(
			screen.getByText(strings.homePage.content.homepageAdTitle)
		).toBeInTheDocument();
		expect(
			screen.getByText(strings.homePage.content.homepageAdText)
		).toBeInTheDocument();
		expect(
			screen.getByText(strings.homePage.content.requestInviteButtonText)
		).toBeInTheDocument();
	});

	it("opens the modal when the button is clicked, and close the modal when click outside the modal area", () => {
		render(
			<QueryClientProvider client={queryClient}>
				<Home />
			</QueryClientProvider>
		);

		// Click the button to open the modal
		fireEvent.click(
			screen.getByText(strings.homePage.content.requestInviteButtonText)
		);

		// Now the modal should be open, and the RequestInviteForm should be rendered
		expect(screen.getByPlaceholderText("Full name")).toBeInTheDocument();
		expect(screen.getByPlaceholderText("Email")).toBeInTheDocument();
		expect(screen.getByPlaceholderText("Confirm Email")).toBeInTheDocument();
		expect(screen.getByText("Send")).toBeInTheDocument();

		// Simulate a click outside the modal area
		fireEvent.click(document.body);

		// Use `queryByText` instead of `getByText` to avoid throwing an error when the element is not found
		const modalContent = screen.queryByText("Your Modal Content");

		// Expect that the modal content is no longer in the document
		expect(modalContent).not.toBeInTheDocument();
	});

	it("matches the snapshot", () => {
		const tree = renderer.create(<Home />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
