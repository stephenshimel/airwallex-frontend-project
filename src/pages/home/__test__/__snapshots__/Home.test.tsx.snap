// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<div
  className="css-wno8gi"
>
  <header
    className="css-izscyd"
    data-testid="header-wrapper"
  >
    <h2
      className="css-0"
    >
      BROCCOLI & CO.
    </h2>
  </header>
  <div
    className="css-11ooj3h"
    data-testid="home-body"
  >
    <h1
      className="css-1qtzou5"
    >
      A better way
    </h1>
    <h1
      className="css-1qtzou5"
    >
      to enjoy every day.
    </h1>
    <h4
      className="css-1bcphut"
    >
      Be the first to know when we launch
    </h4>
    <button
      className="css-9vvphc"
      disabled={false}
      onClick={[Function]}
    >
      Request an invite
    </button>
  </div>
  <footer
    className="css-1cahot1"
    data-testid="footer-wrapper"
  >
    <p
      className="css-eh7lb7"
    >
      Made with ♥ in Melbourne
    </p>
    <p
      className="css-eh7lb7"
    >
      © 2016 Broccoli & Co. All rights reserved
    </p>
  </footer>
</div>
`;
